useStore.js

```diff
@@ -98,6 +98,34 @@ export const useListItemsStore = create(
           listItems: newListItems,
         });
       },
+      updateListItemIndex: (destination, source) => {
```

Comment:
Here you are using "Strategy no. 2": Adding an additional prop called index to your items and then 1. updating the additional prop and 2. swapping items and therefore "Strategy no. 1" swapping the actual index of the items in the list.